Results: add text for the rest of sections (results, discussion and methods) (#18)
greenelab/manubot-gpt-manuscript: content/04.results.md

We also observed a high quality revision of the introdution of PhenoPLIER.
For one paragraph, however, the model failed to keep the format of citations.
Additionally, the model did not converge to a revised text for the last paragraph, and our tool left the error message as an HTML comment at the top of it: `The AI model returned an empty string`.
We observed this issue when debugging the prompts, and it could be related to the fact that the paragraph is large and has a more complex structure than the rest.
However, since the model is stochastic, this can be solved by running the automated revision again.

### Results

Below is a paragraph of the Results section of CCC describing a set of simulations using different datasets shown in a figure (which is Figure 1 in [@doi:10.1101/2022.06.15.496326]).
The figure shows four different datasets with two variables each, and different relationships or patterns named random/independent, non-coexistence, quadratic, and two-lines.

![
](images/diffs/results/ccc-paragraph-01.svg "Diffs - CCC results paragraph 01"){width="100%"}

> In addition to having less sentences that are slightly larger, the revised paragraph consistently uses only the past tense, whereas the original one has a mix of past and present tenses.

Reviewer: ```suggestion
In addition to having fewer sentences that are slightly longer, the revised paragraph consistently uses only the past tense, whereas the original one has tense shifts.
```

The revised paragraph also kept all citations, which although is not explicitely mentioned in the prompts for this section (as it is for introductions), in this case is important.
Math was also kept in the original LaTeX format and the figure was correctly referenced using the Manubot syntax.
The model retained the order of the descriptions of the different relationships in the figure (random/independent, non-coexistence, quadratic, and two-lines), which in this case is desirable since it is the same order as in the figure.
In the third sentence of the revised paragraph (line 3), the model generated a good summary of how all coefficients performed in the last two, nonlinear patterns, and why CCC was able to capture them.
We, as human authors, would make a single change by the end of this sentence to avoid repeating the word "complexity": *"..., while CCC increased the complexity of the model ~~by using different degrees of complexity~~ to capture the relationships"*.
Since a good summary of the performance of all coefficients was already provided, the next two sentences simply describe what the figure shows while keeping a focus on how CCC works.

> It is remarkable how the model mixed the ideas in the last sentences in the original paragraph (lines 4 to 8) to generate three new ones (lines 3 to 5) with the same meaning but in a more concise and clear way.

Reviewer: ```suggestion
In this case study, we found it remarkable that the model mixed the ideas in the last sentences in the original paragraph (lines 4 to 8) to generate three new ones (lines 3 to 5) with the same meaning but more concisely and clearly.
```

The model also produced high-quality revisions for several other paragraphs that would only need minor changes.

Other paragraphs in CCC, however, needed more changes before being ready to be incorporated into the manuscript.
For instance, for some paragraphs, the model generated a revised text that is shorter, more direct and clear.
However, important details were removed, and sometimes sentences changed the meaning.
In this case, we could accept the simplified sentence structure but add back the missing details.

In PhenoPLIER, the model also produced high-quality revisions for most paragraphs, while keeping citations and references to figures, tables and other section of the manuscript in the Manubot/Markdown format.

> In some cases, important details were left out, but they could be easily added back while keeping the better sentence structure suggested.

Reviewer: ```suggestion
In some cases, important details were left out, but they could be easily added back while keeping the improved sentence structure of the revised version.
```

Other cases clearly showed the limitations of revising one paragraph at a time without considering the rest of the text.
An example of this is when we describe our CRISPR screening approach to assess whether top genes in an LV could represent good therapeutic targets:

![
](images/diffs/results/phenoplier-paragraph-01.svg "Diffs - PhenoPLIER results paragraph 01"){width="100%"}

> The model generated a paragraph that has a completely different meaning.

Reviewer: ```suggestion
In one case, the model generated a paragraph with a completely different meaning.
```

> The revised paragraph describes an experiment that does not exist with a reference to a nonexisting section.

Reviewer: ```suggestion
The revised paragraph describes an experiment that does not exist and references a nonexisting section.
```

There is no mention of the CRISPR screen and the gene symbols identified to be associated with the regulation of lipids, which are the key elements in the original text.
Instead, the model seemed to have focused more on the title and keywords of the manuscript (Table @tbl:manuscripts) that are part of every prompt (Figure @fig:ai_revision).
It included the idea of a "gene co-expression" analysis (a keyword) to identify "therapeutic targets" (another keyword), and replaced the mention of "sets of genes" in the original text with "clusters of genes" in the revision (closer to the "clustering" keyword).

> Although this is clearly a bad revision, the output of the model also suggests that the original paragraph is maybe too short or disconnected from the rest, and that it should probably be merged with the next one (which describes follow-up and related experiments).

Reviewer: ```suggestion
Although this was a poor model-based revision, the output suggests that the original paragraph may be too short or disconnected from the rest and that it could be merged with the next one (which describes follow-up and related experiments).
Paragraphs that AI-based models struggle to revise may also reveal paragraphs that would challenge human readers.
```

### Discussion

> All of the paragraphs in the Discussion sections of CCC and PhenoPLIER had a high quality revision.

Reviewer: ```suggestion
In both the CCC and PhenoPLIER manuscripts, revisions to the discussion section appeared to be of high quality.
```

The model kept the correct format when necessary (e.g., using italics for gene symbols), maintained most of the citations, and improved the readability of the text in general.

> Some paragraphs had minor mistakes that are easy to fix by a human author.

Reviewer: ```suggestion
Revisions for some paragraphs introduced minor mistakes that aa human author could readily fix.
```

Below we show a paragraph of CCC where we discuss how not-only-linear correlation coefficients could potentially impact genetic studies of complex traits:

![
](images/diffs/discussion/ccc-paragraph-01.svg "Diffs - CCC discussion paragraph 01"){width="100%"}

Although some minor changes could be added, we believe the revised text reads better than the original.
It is also interesting how the model understood the format of citations and built more complex structures from it.
For instance, the two articles referenced in lines 2 and 3 in the original text were correctly merged into a single citation block and separated with ";" in line 2 of the revised text.

### Methods

> Prompts for the Methods section were the most challenging to design, especially when it comes to the definition of equations.

Reviewer: ```suggestion
Prompts for the Methods section were the most challenging to design, especially when the sections included equations.
```

The prompt for Methods (Figure @fig:ai_revision) is more focused in keeping the technical details, which was especially important for PhenoPLIER, whose Methods section contains paragraphs with several mathematical expressions.

We show below the revision of a paragraph in PhenoPLIER that contains two numbered equations:

![
](images/diffs/methods/phenoplier-paragraph-01.svg "Diffs - PhenoPLIER methods paragraph 01"){width="100%"}

The revised text contains very few changes: all the equations, citations and most of the original text was preserved.
However, it is interesting how the model identified a mistake in the original text (line 8) and fixed it in the revision (line 7).

> Indeed, the equation with the univariate model used by PrediXcan (lines 4 to 6 in the original) includes the *true* effect size $\gamma_l$ (`\gamma_l`) instead of the *estimated* one $\hat{\gamma}_l$ (`\hat{\gamma}_l`).

Reviewer: Am I reading this correctly that this change from GPT-3 is correct? Below, it looks to me like it's doing the opposite unless I'm misunderstanding.

Author: Yes, the GPT-3 model is doing the right thing: it corrected a wrong reference to a mathematical symbol.

Author: I tried to improve the clarity here by splitting the paragraph.

In PhenoPLIER, the model failed to revise one large paragraph with several equations, although it performed relatively well in revising the rest of the section.
Overall, the revision of this section in CCC was good, with some minor and easy-to-fix issues as in the other sections.

> We found again the problem of revising one paragraph at a time without context.

Reviewer: ```suggestion
We also observed issues from revising one paragraph at a time without context.
```
